.github/workflows/build-and-push-containers.yml

```diff
@@ -11,65 +11,44 @@ jobs:
-          MINOR="${MAJOR}.$(echo ${TAG} | cut -d. -f2)"
-          PATCH="${PREFIX}:${TAG}"
-          echo "::set-output name=tags::[\"${MAJOR}\",\"${MINOR}\",\"${PATCH}\"]"
-
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Setup QEMU
-        uses: docker/setup-qemu-action@v1
-
       - name: Setup Docker Buildx
         uses: docker/setup-buildx-action@v1
+        with:
+          install: true
 
       - name: Login to GitHub Container Registry
-        if: ${{ env.DOCKER_USER }}
+        if: ${{ github.event_name == 'release' }}
         uses: docker/login-action@v1
         with:
           registry: ghcr.io
-          username: ${{ env.DOCKER_USER }}
-          password: ${{ secrets.CONTAINER_PAT }}
-
-      - name: Build and push for release
-        if: ${{ github.event_name == 'release' }}
-        uses: docker/build-push-action@v2
-        with:
-          context: .
-          file: ./Dockerfile
-          platforms: linux/amd64
-          pull: true
-          push: true
-          tags: ${{ join(fromJSON(steps.tags.outputs.tags), ',') }}
-          cache-from: type=registry,ref=ghcr.io/laminas/laminas-continuous-integration-action:build-cache
-          cache-to: type=registry,ref=ghcr.io/laminas/laminas-continuous-integration-action:build-cache,mode=max
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build for local pull request
-        if: ${{ github.event_name == 'pull_request' && env.DOCKER_USER }}
-        uses: docker/build-push-action@v2
+      - name: Docker meta
+        id: docker_meta
+        uses: docker/metadata-action@v3
         with:
-          context: .
-          file: ./Dockerfile
-          platforms: linux/amd64
-          pull: true
-          push: false
-          cache-from: type=registry,ref=ghcr.io/laminas/laminas-continuous-integration-action:build-cache
-          cache-to: type=registry,ref=ghcr.io/laminas/laminas-continuous-integration-action:build-cache,mode=max
-
-      - name: Build for fork pull request
-        if: ${{ github.event_name == 'pull_request' && ! env.DOCKER_USER }}
+          images: ghcr.io/laminas/laminas-continuous-integration
+          tags: |
+            type=semver,pattern={{version}}
+            type=ref,event=branch
+            type=ref,event=pr
+          flavor: |
+            latest=false
+
+      - name: Build image. Push for release
         uses: docker/build-push-action@v2
         with:
           context: .
           file: ./Dockerfile
-          platforms: linux/amd64
+          # target: multi-stage build target name
           pull: true
-          push: false
-          cache-from: type=registry,ref=ghcr.io/laminas/laminas-continuous-integration-action:build-cache
+          push: ${{ github.event_name == 'release' }}
+          tags: |
+            ${{ steps.docker_meta.outputs.tags }}
+          labels: ${{ steps.docker_meta.outputs.labels }}
```

Comment:
Is this fetched from the `LABEL` blocks in `Dockerfile`? If so, I can totally see the value in using the `docker/metadata-action`, although it's weird that these aren't simply fetched from the `LABEL` blocks in this step directly :thinking:

Reply:
No, those are populated from action context, you can see output in log for metadata step:

```
Docker tags
  ghcr.io/laminas/laminas-continuous-integration:pr-76
Docker labels
  org.opencontainers.image.title=laminas-continuous-integration-action
  org.opencontainers.image.description=GitHub Action for running a QA check
  org.opencontainers.image.url=https://github.com/laminas/laminas-continuous-integration-action
  org.opencontainers.image.source=https://github.com/laminas/laminas-continuous-integration-action
  org.opencontainers.image.version=pr-76
  org.opencontainers.image.created=2022-01-25T00:22:31.500Z
  org.opencontainers.image.revision=2b0bbcc594923f10ac377486321ba0e1fd8cc366
  org.opencontainers.image.licenses=BSD-3-Clause
```

Reply:
`LABEL` in dockerfile are already applied and those don't need to be specified at runtime